dev/tasks/cleanup-test-resources

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPO_ROOT="$(git rev-parse --show-toplevel)"
+cd ${REPO_ROOT}
+
+GCP_PROJECT_ID=$(gcloud config get project)
```

Comment:
The `kcctest-` convention could make this harder to maintain or left some unreacable projects.
I like that we use labels below when creating project, wondering if we can filter by label here and only delete the matching project.

Reply:
We don't want to delete a project from another test...

I included the date so that we can relatively easily delete old projects.  If we want to we can add a label with the timestamp, but I think we should OSS the cleanup logic first.  (And we should think about whether we're rebuilding boskos and should just use that)